src/rafimage.cpp

```diff
@@ -364,17 +364,25 @@ namespace Exiv2 {
         // sanity check.  Does tiff lie inside the file?
         enforce(Safe::add(tiffOffset, tiffLength) <= io_->size(), kerCorruptedMetadata);
 
-        DataBuf  tiff(tiffLength);
         if (io_->seek(tiffOffset, BasicIo::beg) != 0) throw Error(kerFailedToReadImageData);
-        io_->read(tiff.pData_, tiff.size_);
 
-        if (!io_->error() && !io_->eof())
+        // check if this really is a tiff (some older models just embed a raw bitstream)
+        io_->read(readBuff, 4);
```

Comment:
Looking at this read call, I've a question for @kevinbackhouse. Should all of our `read` calls become [`readOrThrow`](https://github.com/Exiv2/exiv2/pull/1790/files#diff-2831d8bda7242391cc6c1b0462a30ce372b3c56e083145b225cbc33dbbc4011fR99)?  

I guess if we really try to protect exiv2 even from malicious files, then it would be the safest approach, no?

Reply:
@hassec: yes, I would like to migrate to `readOrThrow` wherever possible. I think incomplete reads are almost always caused by a corrupt input file. I think [this `read`](https://github.com/Exiv2/exiv2/blob/3575a8258e101c1294d9f8d229bae92dd600911f/src/jpgimage.cpp#L1293) is the only one that I have come across so far where an incomplete read isn't an error.

Reply:
I think this is goof enough for now:

```suggestion
        if (io_->read(readBuff, 4) != 4) { throw Error(kerFailedToReadImageData); }
```